client/src/objects/drawin.rs

```diff
@@ -43,31 +96,71 @@ impl<'lua> Drawin<'lua> {
     fn new(lua: rlua::Context<'lua>, args: Table<'lua>) -> rlua::Result<Drawin<'lua>> {
         let class = class::class_setup(lua, "drawin")?;
         let mut drawins = lua.named_registry_value::<str, Vec<Drawin>>(DRAWINS_HANDLE)?;
-        let drawin = object_setup(lua, Drawin::allocate(lua, class)?)?
+        let mut drawin = object_setup(lua, Drawin::allocate(lua, class)?)?
             .handle_constructor_argument(args)?
             .build();
+        drawin.create_shell()?;
+        drawin
+            .drawable()?
+            .set_associated_data::<Drawin>("drawin", drawin.clone())?;
         drawins.push(drawin.clone());
         lua.set_named_registry_value(DRAWINS_HANDLE, drawins.to_lua(lua)?)?;
         Ok(drawin)
     }
 
+    pub(crate) fn refresh_pixmap(&mut self, buffer: &Buffer, geometry: Area) -> rlua::Result<()> {
+        {
+            let mut state = self.state_mut()?;
+            let shell = state.surface.as_mut().unwrap();
+            shell.set_size(geometry.size);
+
+            shell.set_buffer(buffer, geometry.origin);
```

Comment:
Should probably only set the buffer and commit if the size has changed, no? Or is this supposed to handle all updates to the drawin? AFAIK the `refreshed` boolean already covers that though.